File: apps/legacy_vault/legacy_vault/Core/Models/VaultRecord.swift
import Foundation

struct VaultRecord: Identifiable, Codable, Hashable {
    let id: UUID
    var name: String
    var vaultType: VaultType
    var encryptedData: Data
    var salt: Data
    var lastPingDate: Date
    var deadlineDays: Int
    var targetEmails: [String]
    var status: VaultStatus

    enum VaultType: String, Codable {
        case passwords, legal, financial, photos, videos, custom
    }

    enum VaultStatus: String, Codable {
        case active, paused, expired, alert_sent
    }

    init(fromCLC clc: CLCVaultRecord) {
        self.id = UUID(uuidString: clc.id) ?? UUID()
        self.name = clc.name
        self.vaultType = VaultType(rawValue: clc.vaultType) ?? .custom
        self.encryptedData = clc.encryptedData
        self.salt = clc.salt
        self.lastPingDate = clc.lastPingDate
        self.deadlineDays = Int(clc.deadlineDays)
        self.targetEmails = clc.targetEmails.components(separatedBy: ",").filter { !$0.isEmpty }
        self.status = VaultStatus(rawValue: clc.status) ?? .active
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: VaultRecord, rhs: VaultRecord) -> Bool {
        lhs.id == rhs.id
    }
}

File: archive/legacy_quarantine/apps/legacy_vault_android/app/src/main/java/com/sfx/legacyvault/core/models/VaultRecord.kt
package com.sfx.legacyvault.core.models

import java.util.UUID

data class VaultRecord(
    val id: String = UUID.randomUUID().toString(),
    val name: String = "",
    val vaultType: String = "custom",
    val encryptedData: String = "",
    val salt: String = "",
    val lastPingDate: Long = 0L,
    val deadlineDays: Int = 0,
    val targetEmails: List<String> = emptyList(),
    val status: String = "active"
)
